### PR TITLE
RELATED: RAIL-2524 Add support for execution warnings

### PR DESCRIPTION
**Pull request by kandl**
---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by kandl** [comment]
ok to test

File: libs/sdk-backend-spi/src/workspace/execution/results.ts [hunk 1]
```diff
@@ -163,6 +164,29 @@ export interface IResultTotalHeader {
...
+
+    /**
+     * Human readable representation of the execution warning
+     */
+    message: string;
+
+    /**
+     * Execution warning parameters (e.g. when filter was not applied - its ObjRef)
+     */
+    parameters?: (ObjRef | boolean | number | string | null)[];
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/execution/results.ts:187]: How are the parameters related to the message in SPI (in Bear there are some placeholders in the message right?). Is this the API we want for SPI? If so, we should probably comment on this here (like what format do the placeholdres have in the message, etc.)

File: libs/sdk-backend-spi/src/workspace/execution/index.ts [hunk 2]
```diff
@@ -315,6 +315,11 @@ export interface IDataView {
      */
     readonly result: IExecutionResult;
 
+    /**
+     * Result warnings.
+     */
```
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/execution/index.ts:320]: please add more contractual info here. something like. 

Backend MAY pass warnings to indicate that the result is different compared to what the caller may expect. For instance the caller executes a definition with particular filter; backend determines that the filter does not have any impact on the computation; backend runs the execution and communicates via warning the the filter was useless.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/410/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1993/ : SUCCESS in 9m 29s

File: libs/sdk-backend-spi/src/workspace/execution/results.ts (hunk 1 above)
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/execution/results.ts:187]: a slightly more detail here. does the message contain placeholders ('%s') which should be replaced by client with the values of parameters?

File: libs/sdk-backend-spi/src/workspace/execution/results.ts [hunk 3]
```diff
@@ -163,6 +164,30 @@ export interface IResultTotalHeader {
...
+ */
+export interface IResultWarning {
+    /**
+     * Unique identifier of the execution warning
+     */
+    warningCode: string;
+
+    /**
+     * Human readable representation of the execution warning, with C-like printf parameter placehodlers
+     * Example: "metric filter on dimension [%s] not applied"
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/execution/results.ts:181]: I'm still missing some relation to the `parameters`. What about adding soemthing like:

> The values for the placeholders are in the `parameters` array in the order they should be used in the placeholders.

?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/412/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1996/ : SUCCESS in 9m 15s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/259/ : SUCCESS in 25s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/514/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1999/ : SUCCESS in 10m 42s

**Pull request merged**